c/datasetjson.c

```diff
@@ -2026,10 +2029,36 @@ void respondWithDatasetMetadata(HttpResponse *response) {
       respondWithError(response, HTTP_STATUS_BAD_REQUEST,"Malformed resume dataset name");
     }
   }
+  
+  if(addQualifiersArg != NULL) {
+    int addQualifiers = !strcmp(addQualifiersArg, "true");
+#define DSN_MAX_LEN 44
+    char dsnNameNullTerm[DSN_MAX_LEN + 1] = {0}; //+1 for null term
+    memcpy(dsnNameNullTerm, dsnName.value, sizeof(dsnName.value));
+    nullTerminate(dsnNameNullTerm, sizeof(dsnNameNullTerm) - 1);
+    if (addQualifiers && dsnLen <= DSN_MAX_LEN) {
+      int asteriskPos = lastIndexOf(dsnNameNullTerm, dsnLen,'*');
+      int dblAsteriskPos = indexOfString(dsnNameNullTerm, dsnLen, "**", 0); //"**" is only valid at the end of a query, cannot appear in the middle of a search
+      int periodPos = lastIndexOf(dsnNameNullTerm, dsnLen, '.');
+      if (asteriskPos < 0 && dblAsteriskPos < 0) {
+        if(dsnLen <= DSN_MAX_LEN - 3 && (periodPos < 0 || periodPos != dsnLen - 1)){ //Query in form of hlq1.hlq2
+          snprintf(dsnNameNullTerm, DSN_MAX_LEN + 1, "%s.**", dsnNameNullTerm);
+        }
+      } else {
+        if (dsnLen <= DSN_MAX_LEN - 3 && asteriskPos == dsnLen - 1 && periodPos != asteriskPos - 1 && dblAsteriskPos < 0) { //query in form of hlq1.hlq2*
+          snprintf(dsnNameNullTerm, DSN_MAX_LEN + 1, "%s.**", dsnNameNullTerm);
```

Comment:
BTW, if you have `FOO.BAR.*` or `FOO.BAR*`, do you actually need to append anything? Have you checked what this returns in that case?